### ISSUE TITLE
Molecular Representation

### ISSUE THREAD

**Issue by Phuangji** [zjunlp/MolGen#21]
Hi,

I want to use this model for molecular representation, how does this model perform in molecular representation? How do I achieve this?
It is worth noting that I tried to use the code example on Huggingface and got exactly the same representation of the individual molecules.
Best regards,
```def smiles_to_embedding(smiles, model, tokenizer, device='cpu'):
    inputs = tokenizer([smiles], return_tensors="pt", padding=True, truncation=True)
    inputs = {k: v.to(device) for k, v in inputs.items()}
    with torch.no_grad():
        outputs = model(input_ids=inputs['input_ids'], attention_mask=inputs['attention_mask'], decoder_input_ids=None)
        hidden_states = outputs.encoder_last_hidden_state  # shape: [batch_size=1, sequence_length, hidden_size]

    embedding = torch.mean(hidden_states, dim=1).squeeze().cpu().numpy()  # shape: [hidden_size]
    print(hidden_states)
    return embedding

device = 'cuda' if torch.cuda.is_available() else 'cpu'
tokenizer = AutoTokenizer.from_pretrained("../MolGen-large")
model = AutoModelForSeq2SeqLM.from_pretrained("../MolGen-large").to(device)

**Comment by Fangyinfff** [comment]
Hi,

Thanks for your attention. I have tested the code and did not encounter the issue of getting the same representation for individual molecules. However, I noticed that your input might be incorrect. For our model, the input should be in the **SELFIES** format.

Try the following code instead:
```
inputs = tokenizer("[C][=C][C][=C][C][=C][Ring1][=Branch1]", return_tensors="pt")
```

Best,

**Comment by Phuangji** [comment]
Thanks for your reply! By the way, when I transform the SMILES to SELFIES, there are some can not be transformed, such as:
C=CCCO[P@TB12](=O)(O)O
The transformation code is ```selfies = sf.encoder(normalized_smiles)
How do I handle this?
Thanks again
Best regards,

**Comment by Fangyinfff** [comment]
I recommend reaching out to the developers of the SELFIES library (https://github.com/aspuru-guzik-group/selfies) for more guidance, as the transformation process depends on the implementation of the SELFIES library.

**Comment by Phuangji** [comment]
Thanks for your kind reply!
